### PR TITLE
#52491 Cosmetic: fix type cast so that it matches return value type

### LINKED ISSUE #52491: Double conversion of value in hashfuncs.h
### Godot version

Not needed

### System information

Not needed

### Issue description

https://github.com/godotengine/godot/blob/d308a0a148bb18114c9ea81410d6b3ed08469e8a/core/templates/hashfuncs.h#L77-L85

in last return value is converted from `uint64` to `int` and at the end to `uint32` which cause to show this error in undefined sanitizer
```
core/templates/hashfuncs.h:85:9: runtime error: implicit conversion from type 'int' of value -399682699 (32-bit, signed) to type 'uint32_t' (aka 'unsigned int') changed the value to 3895284597 (32-bit, unsigned)
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior core/templates/hashfuncs.h:85:9 in 
```
Probably `(int)v` should be changed directly to `uint32_t(v)`

### Steps to reproduce

Not needed

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by kdiduk**
As the function returns "uint32_t", we need to cast returning value to "uint32_t" instead of "int".
A functional cast expression is used, i.e. `uint32_t(value)`.

*Bugsquad edit:* Fixes #52491.

### CODE REVIEW

**Comment by akien-mga** [comment]
The fix looks fine, but please squash the commits together. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

File: core/templates/hashfuncs.h [hunk 1]
```diff
@@ -74,6 +74,13 @@ static inline uint32_t hash_djb2_one_32(uint32_t p_in, uint32_t p_prev = 5381) {
 	return ((p_prev << 5) + p_prev) + p_in;
 }
 
+/**
+ * Thomas Wang's 64-bit to 32-bit Hash function:
+ * http://web.archive.org/web/20071223173210/http:/www.concentric.net/~Ttwang/tech/inthash.htm
```
Comment (akien-mga) [inline core/templates/hashfuncs.h:79]: Both URLs can use HTTPS:
```suggestion
 * https://web.archive.org/web/20071223173210/https:/www.concentric.net/~Ttwang/tech/inthash.htm
```

Comment (kdiduk) [reply core/templates/hashfuncs.h]: Thank you! Done.

**Comment by kdiduk** [comment]
> The fix looks fine, but please squash the commits together. See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

Thank you for the hint. I've updated the PR. Please, review.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**